device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -668,6 +685,12 @@ public void subscribeToDeviceMethod(DeviceMethodCallback deviceMethodCallback, O
         this.subscribeToMethodsInternal(deviceMethodCallback, deviceMethodCallbackContext, deviceMethodStatusCallback, deviceMethodStatusCallbackContext);
     }
 
+    // Used by multiplexing clients to signal to this client what kind of multiplexing client is using this device client
```

Comment:
I am not really clear what this means; is this set only by a multiplexing client, to indicate to this device client's DeviceIO layer (or someone else) that it is a multiplexed client?

Reply:
It is called only by the multiplexing client that "owns" this client, yes. The value set here is used in some if cases when a user tries to set options that are only available to non-multiplexed devices for example

Reply:
It is also used to differentiate between what the old TransportClient allows vs what the new MultiplexingClient allows, hence why this enum has separate values for those two cases.